tools/password_remover.py

```diff
@@ -23,6 +24,112 @@ def is_valid_json(invalid_json):
+        parsed_json = json.loads(valid_json)
+
+        tag_userdata_in_server_json(parsed_json)
+
+        formatted_json_string = json.dumps(parsed_json, indent=4)
+        return formatted_json_string
+
+    except Exception as e:
+        msg = "Exception trying to tag config user data in {0}.  Exception: {1}".format(json_text, e)
+        if log_json_parsing_exceptions:
+            print(msg)
+            traceback.print_exc()
+        return '{"Error":"Error in sgcollect_info password_remover.py trying to tag config user data.  See logs for details"}'
+
+
+def tag_userdata_in_server_json(config):
+        """
+        Given a dictionary that contains a full set of configuration values:
+        - Tag any sensitive user-data fields with <ud></ud> tags.
+        """
+
+        if "databases" in config:
+            dbs = config["databases"]
+            for db in dbs:
+                tag_userdata_in_db_json(dbs[db])
+
+
+def tag_userdata_in_db_config(json_text, log_json_parsing_exceptions=True):
+    """
+    Content postprocessor that tags user data in a db config ready for post-process redaction
+    """
+    try:
+
+        # lower case everything so that "databases" works as a key even if the JSON has "Databases"
+        # as a key.  seems like there has to be a better way!
+        json_text = json_text.lower()
+        valid_json = convert_to_valid_json(json_text)
+        parsed_json = json.loads(valid_json)
+
+        tag_userdata_in_db_json(parsed_json)
+
+        formatted_json_string = json.dumps(parsed_json, indent=4)
+        return formatted_json_string
+
+    except Exception as e:
+        msg = "Exception trying to tag db config user data in {0}.  Exception: {1}".format(json_text, e)
+        if log_json_parsing_exceptions:
+            print(msg)
+            traceback.print_exc()
+        return '{"Error":"Error in sgcollect_info password_remover.py trying to tag db config user data.  See logs for details"}'
+
+
+def tag_userdata_in_db_json(db):
+        """
+        Given a dictionary that contains a set of db configuration values:
+        - Tag any sensitive user-data fields with <ud></ud> tags.
+        """
+
+        if "username" in db:
+            db["username"] = UD(db["username"])
+
+        if "users" in db:
+            users = db["users"]
+            for user in users:
+                if "name" in users[user]:
+                    users[user]["name"] = UD(users[user]["name"])
+                if "admin_channels" in users[user]:
+                    admin_channels = users[user]["admin_channels"]
+                    for i, _ in enumerate(admin_channels):
+                        admin_channels[i] = UD(admin_channels[i])
+                if "admin_roles" in users[user]:
+                    admin_roles = users[user]["admin_roles"]
+                    for i, _ in enumerate(admin_roles):
+                        admin_roles[i] = UD(admin_roles[i])
+            for k, _ in users.items():
```

Comment:
What about just re-using `i` here?  I don't think it's in scope anywhere.

Or maybe use `j` .. a little more obvious it's an index.